src/materialized/tests/sql.rs

```diff
@@ -865,16 +865,13 @@ fn test_tail_table_rw_timestamps() -> Result<(), Box<dyn Error>> {
         let second_rows_verified = verify_rw_pair(&second_rows, "second");
 
         if first_rows_verified && second_rows_verified {
-            // Closing open write timestamp only advances time 1 unit
             let first_write_ts = first_rows[0].get::<_, MzTimestamp>("mz_timestamp");
             let first_closed_ts = first_rows[1].get::<_, MzTimestamp>("mz_timestamp");
-            assert_eq!(first_closed_ts.0 - first_write_ts.0, 1);
+            assert!(first_write_ts < first_closed_ts);
 
-            // We cannot make as strong of claims about the second set of timestamps
-            // because we let more time lapse by having an interceding `FETCH`.
             let second_write_ts = second_rows[0].get::<_, MzTimestamp>("mz_timestamp");
             let second_closed_ts = second_rows[1].get::<_, MzTimestamp>("mz_timestamp");
-            assert!(first_write_ts < second_write_ts);
+            assert!(first_closed_ts <= second_write_ts);
```

Comment:
How can these be equal if verify_rw_pair above verifies that there's a mz_progress row?

Reply:
The `mz_progress` row indicates that you will never see another timestamp [less than the progressed row](https://materialize.com/docs/sql/tail/#progress):

> (`mz_progressed`) is true if no more timestamps will appear that are strictly less than the timestamp.
> The receipt of the explicit progress message at timestamp 4 implies that there are no more updates for either timestamp 2 or 3—but that there may be more data arriving at timestamp 4.

I also found it very unintuitive when fiddling with the `TAIL` docs. Nikhil suggested I pronounce "progressed" as "pruh-gressed" to see if that helped develop a better sense of what was happening.

Reply:
Oops, I misread the second var as the second closed ts, not second write ts. Move along.